src/thor/bidirectional_astar.cc

```diff
@@ -714,24 +796,23 @@ bool BidirectionalAStar::SetForwardConnection(GraphReader& graphreader, const BD
     c = pred.cost().cost + oppcost + opp_pred.transition_cost().cost;
   }
 
-  // Keep the best ones at the front all others to the back
-  best_connections_.emplace_back(CandidateConnection{pred.edgeid(), oppedge, c});
-
-  if (c < best_connections_.front().cost)
-    std::swap(best_connections_.front(), best_connections_.back());
-
   // Set thresholds to extend search
-  if (cost_threshold_ == std::numeric_limits<float>::max()) {
-    float sortcost = std::max(pred.sortcost() + cost_diff_, opp_pred.sortcost());
+  if (cost_threshold_ == std::numeric_limits<float>::max() || c < best_connections_.front().cost) {
     if (desired_paths_count_ == 1) {
-      cost_threshold_ = sortcost + kThresholdDelta;
+      cost_threshold_ = c + kThresholdDelta;
```

Comment:
correct cost threshold. it guarantees that the optimal route will not be missed

Reply:
looks like we can skip `kThresholdDelta` in case of `ignore_hierarchy_limits=true` (no hierarchy prunings, no shortcuts). But it should be tested on some big bicycle/pedestrian dataset that I don't have right now. It can be done in a separate PR

Reply:
hm, on my local planet I didn't notice any difference on `pedestrian_roundabout_routes.txt`, `bicycle_routes.txt`, and `pedestrian_routes.txt` from `test_requests` folder